samples/bot-calling-meeting/csharp/Source/CallingBotSample/Bots/MessageBot.cs

```diff
@@ -256,17 +222,128 @@ private async Task SendResponse(ITurnContext<IMessageActivity> turnContext, stri
             }
         }
 
-        private async Task SendMeetingActionsCard(ITurnContext turnContext, CancellationToken cancellationToken)
+        private async Task<TResult> MakeGraphCallThatMightNotBeFound<TResult>(Func<Task> function, string? callId, Func<string, Task<TResult>> errorHandler)
         {
-            var conversationStateAccessors = conversationState.CreateProperty<MeetingActionDetails>(nameof(MeetingActionDetails));
-            var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new MeetingActionDetails());
+            if (callId == null)
+            {
+                // Without the Meeting ID we are unable to play the prompt
+                //await turnContext.SendActivityAsync("Meeting ID not found, please use the 'Create call' button to create the call.");
+                return await errorHandler("Meeting ID not found, please use the 'Create call' button to create the call.");
+            }
+
+            try
+            {
+                await function();
+            }
+            catch (ServiceException ex)
+            {
+                if (ex.StatusCode != HttpStatusCode.NotFound)
+                {
+                    // If it's not a NotFound error please ignore
+                    throw ex;
+                }
 
-            var resourceResponse = await turnContext.SendActivityAsync(MessageFactory.Attachment(adaptiveCardFactory.CreateMeetingActionsCard()));
-            conversationData.MeetingActionsCardActivityId = resourceResponse.Id;
-            await conversationState.SaveChangesAsync(turnContext, false, cancellationToken);
+                logger.LogError("Call not found. Return error");
+                return await errorHandler("That action failed. Unable to find call");
+            }
+
+            return default;
         }
 
-        private TaskModuleResponse CreateTaskModuleResponse(string value)
+        private async Task<Call> JoinScheduledMeeting(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var users = await TeamsInfo.GetPagedMembersAsync(turnContext, cancellationToken: cancellationToken);
+
+            return await callService.Create(
+                new ChatInfo { ThreadId = turnContext.Activity.Conversation.Id },
+                new OrganizerMeetingInfo
+                {
+                    Organizer = new IdentitySet
+                    {
+                        User = new Identity
+                        {
+                            Id = users.Members[0].AadObjectId,
+                            AdditionalData = new Dictionary<string, object> { { "tenantId", azureAdOptions.TenantId } }
```

Comment:
Is it possible to use the tenant ID in the activity?

Reply:
updated